dali/operators/image/resize/resize_attr_test.cc

```diff
@@ -0,0 +1,475 @@
+    CHECK_PARAMS(attr.params_[1], { { 480, 384, 288 }, { 0, 0, 0 }, { 400, 320, 240 } });
+  }
+}
+
+TEST(ResizeAttr, Resize3DSeparateArgs) {
+  ArgumentWorkspace ws;
+  auto zcoord = std::make_shared<TensorList<CPUBackend>>();
+  zcoord->set_pinned(false);
+  zcoord->Resize(TensorListShape<0>(2));
+  *zcoord->mutable_tensor<float>(0) = 140;
+  *zcoord->mutable_tensor<float>(1) = 150;
+  ws.AddArgumentInput("resize_z", zcoord);
+
+  TensorListShape<> shape = {{
+    TensorShape<3>{ 123, 234, 345 },
+    TensorShape<3>{ 321, 432, 543 }
+  }};
+  {
+    OpSpec spec("Resize");
+    spec.AddArg("resize_x", 120.0f);
+    spec.AddArg("resize_y", 130.0f);
+    spec.AddArg("batch_size", shape.num_samples());
+    spec.AddArgumentInput("resize_z", "resize_z");
+
+    ResizeAttr attr;
+    attr.PrepareResizeParams(spec, ws, shape, "DHW");
+
+    CHECK_PARAMS(attr.params_[0], { { 140, 130, 120 }, { 0, 0, 0 }, { 123, 234, 345 } });
+    CHECK_PARAMS(attr.params_[1], { { 150, 130, 120 }, { 0, 0, 0 }, { 321, 432, 543 } });
+  }
+}
+
+TEST(ResizeAttr, Resize3DSubpixelScale) {
+  ArgumentWorkspace ws;
+  auto zcoord = std::make_shared<TensorList<CPUBackend>>();
+  zcoord->set_pinned(false);
+  zcoord->Resize(TensorListShape<0>(2));
+  *zcoord->mutable_tensor<float>(0) = 140;
+  *zcoord->mutable_tensor<float>(1) = 150;
+  ws.AddArgumentInput("resize_z", zcoord);
+
+  TensorListShape<> shape = {{
+    TensorShape<3>{ 123, 234, 345 },
+    TensorShape<3>{ 321, 432, 543 }
+  }};
+
+  // Configure ResizeAttr with missing Y coordinate
+  OpSpec spec("Resize");
+  spec.AddArg("resize_x", 120.0f);
+  spec.AddArg("batch_size", shape.num_samples());
+  spec.AddArgumentInput("resize_z", "resize_z");
+
+  ResizeAttr attr;
+  attr.PrepareResizeParams(spec, ws, shape, "DHW");
+
+  // Y coordinate is missing and mode is "default" - use geometric mean of the xz scales for the
+  // missing coordinate.
+  double h0 = 234;
+  double h1 = 432;
+  // Calculate the ideal, subpixel, outptut shape.
+  double subpixel_y0 = h0 * sqrt(140.0/123 * 120 / 345);
+  double subpixel_y1 = h1 * sqrt(150.0/321 * 120 / 543);
+  // Real shape needs to be integral - let's round it.
+  int y0 = std::roundl(subpixel_y0);
+  int y1 = std::roundl(subpixel_y1);
+
+  // Now let's adjust the input ROI. It will be scaled proprtionately to the ratio
+  // of actual (rounded) size to the ideal size.
+  double center0 = h0 * 0.5;
+  double center1 = h1 * 0.5;
+  float lo0 = center0 - center0 * y0 / subpixel_y0;
+  float hi0 = center0 + center0 * y0 / subpixel_y0;
+  float lo1 = center1 - center1 * y1 / subpixel_y1;
+  float hi1 = center1 + center1 * y1 / subpixel_y1;
+
+  CHECK_PARAMS(attr.params_[0], { { 140, y0, 120 }, { 0, lo0, 0 }, { 123, hi0, 345 } });
+  CHECK_PARAMS(attr.params_[1], { { 150, y1, 120 }, { 0, lo1, 0 }, { 321, hi1, 543 } });
+
+  // Let's disable subpixel scaling
+  spec.AddArg("subpixel_scale", false);
```

Comment:
Can you add a cease with `subpixel_scale` set to true (even if it is a default value) so it would be easier to check which case checks this. Now I cannot easily tell which of the remaining cases tests that (all, only selected?).

Reply:
Actually, I want to test that it _is_ the default - however, I added a check that `attr.subpixel_scale_` is indeed true/false (see above/below).